drivers/i2c/i2c_smartbond.c

```diff
@@ -126,6 +132,8 @@ static int i2c_smartbond_configure(const struct device *dev, uint32_t dev_config
 
 	key = k_spin_lock(&data->lock);
 
+	i2c_smartbond_pm_policy_state_lock_get(dev);
```

Comment:
Potential deadlock: `i2c_smartbond_configure` is invoked by `i2c_smartbond_resume` which is called following `PM_DEVICE_ACTION_RESUME`. We should first check if the controller is first disabled and then lock pm constraints. Otherwise, it should not be called.

Reply:
Should we use pm_device_runtime_usage() instead to avoid deadlock? E.g.
```
	if (pm_device_runtime_usage(dev) == 0) {
		i2c_smartbond_pm_policy_state_lock_get(dev);
	}
...
	if (pm_device_runtime_usage(dev) == 1) {
		i2c_smartbond_pm_policy_state_lock_put(dev);
	}
```

Reply:
You are right. I introduced a i2c_smartbond_apply_configure() to overcame the issue.